packages/openapi-to-graphql/src/resolver_builder.ts

```diff
@@ -360,131 +367,191 @@ export function getResolver({
 
           if (response.headers['content-type']) {
             /**
-             * If the response body is type JSON, then parse it
+             * Throw warning if the non-application/json content does not
+             * match the OAS.
+             *
+             * Use an inclusion test in case of charset
              *
-             * content-type may not be necessarily 'application/json' it can be
-             * 'application/json; charset=utf-8' for example
+             * i.e. text/plain; charset=utf-8
              */
-            if (response.headers['content-type'].includes('application/json')) {
-              body = JSON.parse(body)
+            if (
+              !(
+                response.headers['content-type'].includes(
+                  operation.responseContentType
+                ) ||
+                operation.responseContentType.includes(
+                  response.headers['content-type']
```

Comment:
Do we assume header keys are already normalized? E.g., what would happen if the header key is `Content-type`?

Reply:
No, I do not think we can assume. See [here](https://github.com/request/request#responsecaselessgetheader-name). I will make the necessary changes.

Reply:
Oh wait, I think this is good. I looked at the TS declarations.

Reply:
All HTTP headers used in this way are lowercased.